tests/trainer/test_trainer_tricks.py

```diff
@@ -112,33 +112,46 @@ def test_overfit_batch_limits(tmpdir):
     # ------------------------------------------------------
     for split in (RunningStage.VALIDATING, RunningStage.TESTING):
 
+        # ------------------------------------------------------
+        # test overfit_batches action
+        # ------------------------------------------------------
+
         # ------------------------------------------------------
         # test overfit_batches as percent
         # ------------------------------------------------------
         trainer = Trainer(overfit_batches=0.11)
         trainer._data_connector.attach_dataloaders(model)
-        loader_num_batches, dataloaders = trainer._reset_eval_dataloader(split, model=model)
-        assert loader_num_batches[0] == num_train_samples
-
-        # make sure we turned off shuffle for the user
-        assert isinstance(dataloaders[0].sampler, SequentialSampler)
-
-        # make sure the loaders are the same
-        (xb, yb) = next(iter(dataloaders[0]))
-        assert torch.eq(xa, xb).all()
-        assert torch.eq(ya, yb).all()
+        loader_num_batches, _ = trainer._reset_eval_dataloader(split, model=model)
+        if split == RunningStage.VALIDATING:
+            assert loader_num_batches[0] == 0
+        else:
+            assert loader_num_batches[0] == len(test_loader)
 
         # ------------------------------------------------------
         # test overfit_batches as int
         # ------------------------------------------------------
         trainer = Trainer(overfit_batches=1)
         trainer._data_connector.attach_dataloaders(model)
         loader_num_batches, dataloaders = trainer._reset_eval_dataloader(split, model=model)
-        assert loader_num_batches[0] == 1
+        if split == RunningStage.VALIDATING:
+            assert loader_num_batches[0] == 0
+        else:
+            assert loader_num_batches[0] == len(test_loader)
+            # make sure we turned off shuffle for the user
+            assert isinstance(dataloaders[0].sampler, SequentialSampler)
```

Comment:
@rohitgr7 quick question. How should we support overfit_batches with DDP? I think we would need to make sure the DistributedSampler shuffle value is set to False.

Reply:
@tchaton yes we do here: https://github.com/PyTorchLightning/pytorch-lightning/blob/3d6262b7a91215e72019d720e742e6261e1636dc/pytorch_lightning/trainer/data_loading.py#L187

I guess there is no test for it.. I'll add one in a follow-up with some cleanup.